lib/trino-orc/src/test/java/io/trino/orc/TestSliceDictionaryColumnReader.java

```diff
@@ -0,0 +1,135 @@
+        OrcReader orcReader = OrcReader.createOrcReader(dataSource, new OrcReaderOptions())
+                .orElseThrow(() -> new RuntimeException("File is empty"));
+        Footer footer = orcReader.getFooter();
+        List<OrcColumn> columns = orcReader.getRootColumn().getNestedColumns();
+        assertTrue(columns.size() == 1);
+        StripeReader stripeReader = new StripeReader(
+                dataSource,
+                UTC,
+                Optional.empty(),
+                footer.getTypes(),
+                ImmutableSet.copyOf(columns),
+                footer.getRowsInRowGroup(),
+                OrcPredicate.TRUE,
+                ORIGINAL,
+                new OrcMetadataReader(),
+                Optional.empty());
+        AggregatedMemoryContext memoryContext = newSimpleAggregatedMemoryContext();
+        SliceDictionaryColumnReader columnReader = new SliceDictionaryColumnReader(columns.get(0), memoryContext.newLocalMemoryContext(TestSliceDictionaryColumnReader.class.getSimpleName()), -1, false);
+
+        List<StripeInformation> stripeInformations = footer.getStripes();
+        for (StripeInformation stripeInformation : stripeInformations) {
+            Stripe stripe = stripeReader.readStripe(stripeInformation, newSimpleAggregatedMemoryContext());
+            List<RowGroup> rowGroups = stripe.getRowGroups();
+            columnReader.startStripe(stripe.getFileTimeZone(), stripe.getDictionaryStreamSources(), stripe.getColumnEncodings());
+
+            for (RowGroup rowGroup : rowGroups) {
+                columnReader.startRowGroup(rowGroup.getStreamSources());
+                columnReader.prepareNextRead(1000);
+                columnReader.readBlock();
+                // memory usage check
+                assertGreaterThanOrEqual(memoryContext.getBytes(), columnReader.getRetainedSizeInBytes());
+            }
+        }
+
+        // Statistic should more than or equals size of all array members that primary type.
+        columnReader.close();
+        assertTrue(memoryContext.getBytes() == 0);
+    }
+
+    private List<String> createValues()
+    {
+        Random random = new Random();
+        List<String> dictionary = createDictionary(random);
+
+        List<String> values = new ArrayList<>();
+        for (int i = 0; i < ROWS; ++i) {
+            if (random.nextBoolean()) {
+                values.add(dictionary.get(random.nextInt(dictionary.size())));
+            }
+            else {
+                values.add(null);
+            }
+        }
+        return values;
+    }
+
+    private List<String> createDictionary(Random random)
+    {
+        List<String> dictionary = new ArrayList<>();
+        for (int dictionaryIndex = 0; dictionaryIndex < DICTIONARY; dictionaryIndex++) {
+            dictionary.add(randomAsciiString(random));
+        }
+        return dictionary;
+    }
+
+    // this is not appropriate for benchmarking with compression
```

Comment:
add comment why

Reply:
Method `randomAsciiString` is copyed from `BenchmarkColumnReaders#randomAsciiString`, with this comment line.
The original comment probably means that there has `random` procedure, can cause result of benchmark unstable.

This comment line is nomeans in this class, has removed now.